### PR TITLE
(nobug) - Mochitest: messages are rendered in the What's New panel

### PR DESCRIPTION
**Pull request by piatra**
The test opens the panel, clicks Whats New and checks the correct number of messages are rendered.

### CODE REVIEW

**Comment by Mardak** [comment]
Looks like on linux-not-quantum-render, the test is failing:
```
[task 2019-08-30T13:35:40.820Z] 13:35:40     INFO - TEST-PASS | browser/components/newtab/test/browser/browser_asrouter_whatsnewpanel.js | FxA test message exists - 4 == true - 
[task 2019-08-30T13:35:40.821Z] 13:35:40     INFO - TEST-PASS | browser/components/newtab/test/browser/browser_asrouter_whatsnewpanel.js | Panel is visible - true == true - 
[task 2019-08-30T13:35:40.822Z] 13:35:40     INFO - TEST-PASS | browser/components/newtab/test/browser/browser_asrouter_whatsnewpanel.js | What's New is present - true == true - 
[task 2019-08-30T13:35:40.822Z] 13:35:40     INFO - Buffered messages finished
[task 2019-08-30T13:35:40.822Z] 13:35:40     INFO - TEST-UNEXPECTED-FAIL | browser/components/newtab/test/browser/browser_asrouter_whatsnewpanel.js | Uncaught exception - undefined - timed out after 50 tries.
```
https://taskcluster-artifacts.net/OlortGRDROWUHZPZVhOhUQ/0/public/test_info//mozilla-test-fail-screenshot_VGI0xl.png

Screenshot seems to indicate the `whatsNewBtn.click()` doesn't open up the panel.

**Comment by Mardak** [comment]
And windows debug seems to fail at that same `whatsNewBtn.click()` step roughly 1 in 10 times.

File: test/browser/browser_asrouter_whatsnewpanel.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  Object.defineProperty(ToolbarPanelHub, "messages", {
+    get: () => Promise.resolve(msgs),
+    configurable: true,
+  });
+
+  await ToolbarPanelHub.enableAppmenuButton();
+
+  const mainView = document.getElementById("appMenu-mainView");
+  UITour.showMenu(window, "appMenu");
+  await BrowserTestUtils.waitForEvent(mainView, "ViewShown");
```
Comment (Mardak) [inline test/browser/browser_asrouter_whatsnewpanel.js]: Looks like we were clicking too early before the main view was done showing, so internally it was preventing loading of a subpanel.

https://searchfox.org/mozilla-central/rev/c8dba8c886609d0fdb8fae68bab865bf03f1165c/browser/components/customizableui/PanelMultiView.jsm#671-676

**Comment by Mardak** [comment]
Pushed latest for another 10 runs https://treeherder.mozilla.org/#/jobs?repo=try&revision=4319ab0825986d6f990334cbb87fefd94d602ba8

**Comment by piatra** [comment]
Thanks :+1:

**Pull request merged**